src/AgDatabase.cs

```diff
@@ -233,5 +233,10 @@ public void MultiUserMode()
     {
       _listener.Primary.Database(Name).MultiUserMode();
     }
+
+    public void CheckDBConnections(int connectionTimeout)

```

Comment:
This is expected to throw when the first failure is encountered, right? Is that the desired behaviour, or would we want it to check all of the hosts in a cluster and give use the health of each?

Reply:
That was my thinking behind it - if it can't connect to even one it should fail - i.e. raise exception

But we can refactor to parallely check each instance and return a list as a health report - could even be a list of exceptions?

Reply:
For my use-case I just need the former